### PR TITLE
Check invalid access key

### PR DESCRIPTION
**Pull request by lcadzy**

### CODE REVIEW

**Comment by lcadzy** [comment]
#2912

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/SharedKeyHttpObservable.java [hunk 1]
```diff
@@ -52,8 +52,9 @@ public SharedKeyHttpObservable(String accountName, String accessKey) {
         setDefaultHeaderGroup(defaultHeaders);
         try {
             this.cred = new SharedKeyCredential(accountName, accessKey);
-        } catch (InvalidKeyException e) {
-            throw new IllegalArgumentException("Cannot init shared key credential");
+        } catch (Exception ex) {
+            log().warn("Init shared key credential encouter exception" + ex.toString());
```
Comment (lcadzy) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/SharedKeyHttpObservable.java]: typo.change next iteration

Comment (wezhang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/SharedKeyHttpObservable.java]: ‘Can’t create shared key credentials’

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobRemoteProcess.java [hunk 2]
```diff
@@ -226,6 +227,10 @@ protected void ctrlError(String message) {
                 .deploy(artifactPath)
                 .onErrorResumeNext(err -> {
                     Throwable rootCause = err.getCause() != null ? err.getCause() : err;
+                    if (rootCause.getMessage().contains(String.valueOf(HttpStatus.FORBIDDEN_403))) {
+                        return Observable.error(new IllegalArgumentException("Failed to upload Spark application artifacts.The shared key is invalid."));
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobRemoteProcess.java]: Could you leverage the following error handling codes for this case?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/HttpObservable.java [hunk 3]
```diff
@@ -59,6 +59,7 @@
 import org.apache.http.ssl.SSLContextBuilder;
 import org.apache.http.ssl.TrustStrategy;
 import org.apache.http.util.EntityUtils;
+import org.mortbay.jetty.HttpStatus;
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/HttpObservable.java]: What's this?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/SharedKeyHttpObservable.java [hunk 4]
```diff
@@ -52,8 +52,9 @@ public SharedKeyHttpObservable(String accountName, String accessKey) {
         setDefaultHeaderGroup(defaultHeaders);
         try {
             this.cred = new SharedKeyCredential(accountName, accessKey);
-        } catch (InvalidKeyException e) {
-            throw new IllegalArgumentException("Cannot init shared key credential");
+        } catch (Exception ex) {
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/SharedKeyHttpObservable.java]: The `InvalidKeyException` is removed, what types exception will be caught here?

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobRemoteProcess.java [hunk 5]
```diff
@@ -31,6 +31,7 @@
 import com.microsoft.azuretools.azurecommons.helpers.NotNull;
 import com.microsoft.azuretools.azurecommons.helpers.Nullable;
 import org.apache.commons.io.output.NullOutputStream;
+import org.eclipse.jetty.http.HttpStatus;
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobRemoteProcess.java]: What's this?

**Pull request merged**